auto_gptq/modeling/_base.py

```diff
@@ -526,81 +570,101 @@ def save_quantized(
+                            f"After converting safetensors_metadata keys to strings, the key '{new_key}' is duplicated. Ensure that all your metadata keys are strings to avoid overwriting."
+                        )
+                    new_safetensors_metadata[new_key] = new_value
+            safetensors_metadata = new_safetensors_metadata
+            if converted_keys:
+                logger.debug(
+                    f"One or more safetensors_metadata keys or values had to be converted to str(). Final safetensors_metadata: {safetensors_metadata}"
+                )
 
-            # Store the quantization configuration as safetensors metadata
-            from auto_gptq import __version__
+        # Format is required to enable Accelerate to load the metadata
+        # otherwise it raises an OSError
+        safetensors_metadata["format"] = "pt"
 
-            safetensors_metadata["auto_gptq_version"] = str(__version__)
-            safetensors_metadata["gptq_bits"] = str(self.quantize_config.bits)
-            safetensors_metadata["gptq_group_size"] = str(self.quantize_config.group_size)
-            safetensors_metadata["gptq_desc_act"] = str(self.quantize_config.desc_act)
-            safetensors_metadata["gptq_damp_percent"] = str(self.quantize_config.damp_percent)
-            safetensors_metadata["gptq_" + CHECKPOINT_FORMAT_FIELD] = self.quantize_config.checkpoint_format
-            safetensors_metadata["gptq_" + QUANT_METHOD_FIELD] = self.quantize_config.quant_method
+        # Store the quantization configuration as safetensors metadata
+        from auto_gptq import __version__
+
+        safetensors_metadata["auto_gptq_version"] = str(__version__)
+        safetensors_metadata["gptq_bits"] = str(self.quantize_config.bits)
+        safetensors_metadata["gptq_group_size"] = str(self.quantize_config.group_size)
+        safetensors_metadata["gptq_desc_act"] = str(self.quantize_config.desc_act)
+        safetensors_metadata["gptq_damp_percent"] = str(self.quantize_config.damp_percent)
+        safetensors_metadata["gptq_" + CHECKPOINT_FORMAT_FIELD] = self.quantize_config.checkpoint_format
+        safetensors_metadata["gptq_" + QUANT_METHOD_FIELD] = self.quantize_config.quant_method
+
+        safe_save(state_dict, join(save_dir, model_save_name), safetensors_metadata)
 
-            safe_save(state_dict, join(save_dir, model_save_name), safetensors_metadata)
-        else:
-            model_save_name = model_base_name + ".bin"
-            torch.save(self.model.state_dict(), join(save_dir, model_save_name))
 
         self.model.config.quantization_config = self.quantize_config.to_dict()
         self.model.config.save_pretrained(save_dir)
         self.quantize_config.save_pretrained(save_dir)
         self.quantize_config.model_name_or_path = save_dir
         self.quantize_config.model_file_base_name = model_base_name
 
+        # revert model
+        if revert_model is not None:
+            self.model = revert_model
```

Comment:
The modif in `convert_gptq_v1_to_v2_format` is done in place, so this won't work.